fpx-cli/src/data.rs

```diff
@@ -218,4 +220,62 @@ impl Store for LibsqlStore {
 
         Ok(Some(rows_affected))
     }
+
+    async fn settings_upsert(&self, _tx: &Transaction, settings: Settings) -> Result<Settings> {
```

Comment:
Is there a specific reason you didn't just go with a settings struct per row, instead of having each property be a value?

In the upsert handler you could fetch the current value, update from the http payload and then save that object as one whole.

Reply:
i dont understand what you mean by this? having the whole setting struct be a column and then keep saving the settings struct newly?

Reply:
The current implementation stores each value of each struct  of each user as a separate row. We are storing the key as a value in the database. So you have row for [user1, setting_x] and a row for [user1, setting_y].

It is quite common to store the entire setting of a single user in a single row, where each setting property would map to a specific column. So in this situation you'd have three column, 1 for the user id, and two column for setting_x and setting_y.

Reply:
we dont actually store per-user settings here, its global settings so I think this makes more sense, id agree if we had to store per-user settings